lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -190,20 +190,45 @@ class LastConnectResolver : public FIRRTLVisitor<ConcreteT> {
 
   /// For every leaf field in the sink, record that it exists and should be
   /// initialized.
-  void declareSinks(Value value, Flow flow) {
+  void declareSinks(Value value, Flow flow, bool local) {
     auto type = value.getType();
     unsigned id = 0;
 
     // Recurse through a bundle and declare each leaf sink node.
-    std::function<void(Type, Flow)> declare = [&](Type type, Flow flow) {
+    std::function<void(Type, Flow, bool)> declare = [&](Type type, Flow flow,
+                                                        bool local) {
+      // If this is a class type, recurse to each of the fields.
+      if (auto classType = type_dyn_cast<ClassType>(type)) {
+        if (local) {
+          // If this is a local object declaration, then we are responsible for
+          // initializing it's input ports.
+          for (auto &element : classType.getElements()) {
+            id++;
+            if (element.direction == Direction::Out)
+              declare(element.type, flow, false);
```

Comment:
NBD, but noticed not using `/*local=*/` here, is that because we're recursing?

Reply:
Yep! I figure we're not going to loose track of what this false means.